integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/utils/ServerManager.java

```diff
@@ -0,0 +1,206 @@
+     */
+    private static Logger log = Logger.getLogger(ServerManager.class);
+
+    private String carbonHome;
+    private Process process;
+    private ServerLogReader inputStreamHandler;
+    private ServerLogReader errorStreamHandler;
+
+
+    /**
+     * Setup server home by extracting the distribution on temporary directory.
+     *
+     * @param serverZipFilePath distribution zip file path
+     * @return temporary carbon home path after extraction
+     * @throws IOException
+     */
+    public synchronized String setupServerHome(String serverZipFilePath) throws IOException {
+
+        String fileSeparator = File.separator.equals("\\") ? "\\" : "/";
+        if (fileSeparator.equals("\\")) {
+            serverZipFilePath = serverZipFilePath.replace("/", "\\");
+        }
+        int indexOfZip = serverZipFilePath.lastIndexOf(".zip");
+
+        String extractedCarbonDir = serverZipFilePath.substring(serverZipFilePath.lastIndexOf(fileSeparator) + 1,
+                indexOfZip);
+        FileManipulator.deleteDirectory(new File(extractedCarbonDir));
+        String extractDir = "carbontmp" + System.currentTimeMillis();
+        String baseDir = System.getProperty("basedir", ".") + File.separator + "target";
+        log.info("Extracting carbon zip file.. ");
+        (new ArchiveExtractor()).extractFile(serverZipFilePath, baseDir + File.separator + extractDir);
+        this.carbonHome = (new File(baseDir)).
+                getAbsolutePath() + File.separator + extractDir + File.separator + extractedCarbonDir;
+
+        return carbonHome;
+    }
+
+
+    /**
+     * Start server instance
+     *
+     * @param carbonHomePath carbon home path
+     * @throws IOException
+     */
+    public synchronized void startServer(String carbonHomePath) throws IOException {
+
+        File carbonHomeDirectory = new File(carbonHomePath);
+        String scriptName = getStartupScriptFileName(carbonHomePath);
+        String[] cmdArray;
+
+        if (System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows")) {
+            carbonHomeDirectory = new File(carbonHomePath + File.separator + "bin");
+            cmdArray = new String[]{"cmd.exe", "/c", scriptName + ".bat"};
+            process = Runtime.getRuntime().exec(cmdArray, (String[]) null, carbonHomeDirectory);
+        } else {
+            cmdArray = new String[]{"sh", "bin/" + scriptName + ".sh"};
+            process = Runtime.getRuntime().exec(cmdArray, (String[]) null, carbonHomeDirectory);
+        }
+
+        errorStreamHandler = new ServerLogReader("errorStream", process.getErrorStream());
+        inputStreamHandler = new ServerLogReader("inputStream", process.getInputStream());
+        inputStreamHandler.start();
+        errorStreamHandler.start();
+
+        long startupWaitTimeOut = 60000L;
+
+        long time = System.currentTimeMillis() + startupWaitTimeOut;
+
+        while (true) {
```

Comment:
shall we add some polling delay within this loop, otherwise this will consume more processing power?